src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -0,0 +1,2078 @@
+            var model = comp.GetSemanticModel(tree);
+
+            var nodes = tree.GetRoot().DescendantNodes().OfType<AssignmentExpressionSyntax>();
+            Assert.Equal(2, nodes.Count());
+
+            foreach (var assign in nodes)
+            {
+                var node = assign.Left;
+                Assert.Equal("P1", node.ToString());
+                Assert.Equal("System.Int32 C.P1 { get; set; }", model.GetSymbolInfo(node).Symbol.ToTestDisplayString());
+            }
+        }
+
+        [Fact]
+        public void ModReqOnAccessorParameter()
+        {
+            string il = @"
+.class public auto ansi beforefieldinit C extends System.Object
+{
+    .method public hidebysig specialname newslot virtual instance void set_Property ( int32 modreq(System.Runtime.CompilerServices.IsInitOnly) 'value' ) cil managed
+    {
+        IL_0000: ldnull
+        IL_0001: throw
+    }
+
+    .method public hidebysig specialname rtspecialname instance void .ctor () cil managed
+    {
+        IL_0000: ldnull
+        IL_0001: throw
+    }
+
+    .property instance int32 Property()
+    {
+        .set instance void C::set_Property(int32 modreq(System.Runtime.CompilerServices.IsInitOnly))
+    }
+}
+
+.class public auto ansi sealed beforefieldinit System.Runtime.CompilerServices.IsInitOnly extends System.Object
+{
+    .method public hidebysig specialname rtspecialname instance void .ctor () cil managed
+    {
+        IL_0000: ldnull
+        IL_0001: throw
+    }
+}
+";
+
+            string source = @"
+public class Derived : C
+{
+    public override int Property { set { throw null; } }
+}
+public class Derived2 : C
+{
+    public override int Property { init { throw null; } }
+}
+";
+
+            var reference = CreateMetadataReferenceFromIlSource(il);
+            var comp = CreateCompilation(source, references: new[] { reference }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (4,25): error CS0569: 'Derived.Property': cannot override 'C.Property' because it is not supported by the language
+                //     public override int Property { set { throw null; } }
+                Diagnostic(ErrorCode.ERR_CantOverrideBogusMethod, "Property").WithArguments("Derived.Property", "C.Property").WithLocation(4, 25),
+                // (8,25): error CS0569: 'Derived2.Property': cannot override 'C.Property' because it is not supported by the language
+                //     public override int Property { init { throw null; } }
+                Diagnostic(ErrorCode.ERR_CantOverrideBogusMethod, "Property").WithArguments("Derived2.Property", "C.Property").WithLocation(8, 25)
+                );
+        }
+
+        [Fact]
+        public void TestSyntaxFacts()
+        {
+            Assert.True(SyntaxFacts.IsAccessorDeclaration(SyntaxKind.InitAccessorDeclaration));
+            Assert.True(SyntaxFacts.IsAccessorDeclarationKeyword(SyntaxKind.InitKeyword));
+        }
+    }
+}
```

Comment:
Consider adding a few tests where we add an access modifier to the `init` accessor.

Reply:
Added

---
In reply to: [411855029](https://github.com/dotnet/roslyn/pull/43275#discussion_r411855029) [](ancestors = 411855029)